pandas/tests/generic/test_series.py

```diff
@@ -102,23 +102,34 @@ def test_nonzero_single_element(self):
         s = Series([False])
         assert not s.bool()
 
+        msg = "The truth value of a Series is ambiguous"
         # single item nan to raise
         for s in [Series([np.nan]), Series([pd.NaT]), Series([True]),
                   Series([False])]:
-            pytest.raises(ValueError, lambda: bool(s))
+            with pytest.raises(ValueError, match=msg):
+                bool(s)
 
+        msg = "bool cannot act on a non-boolean single element Series"
         for s in [Series([np.nan]), Series([pd.NaT])]:
-            pytest.raises(ValueError, lambda: s.bool())
+            with pytest.raises(ValueError, match=msg):
+                s.bool()
 
         # multiple bool are still an error
+        msg = "The truth value of a Series is ambiguous"
```

Comment:
nitpick: Do we need to define this string 3 times in this file?

Reply:
fine for now...

this is surely an indication of tests that could be refactored to use more pytest idiom... but that'll be for a follow-up. 

also the messages serve as additional documentation for the tests